translator/source_code/cotrip_translator.py

```diff
@@ -0,0 +1,421 @@
+    all_lanes_open = ['No Closure', -'Intermittent Traffic Stops', 'Shoulder Closure', 'Right Shoulder Closed', 'Off-Ramp Closure', 'On-Ramp Closure',  ]
+    alternating_one_way = ['Single Lane Traffic', 'Alternating Traffic' ]
+    unknown = []
+    vehicle_impact = 'unknown'    
+    if closure_type in all_lanes_closed:
+        vehicle_impact = 'all-lanes-closed'
+    elif closure_type in some_lanes_closed:
+        vehicle_impact = 'some-lanes-closed'
+    elif closure_type in all_lanes_open:
+        vehicle_impact = 'all-lanes-open'
+    elif closure_type in alternating_one_way:
+        vehicle_impact = 'alternating-one-way'
+    return vehicle_impact
+
+
+# function to parse polyline to geometry line string
+#-104.480110,37.007645,0 -104.480103,37.008034,0 -104.480125,37.008469,0
+def parse_polyline(polylinestring):
+    if not polylinestring or type(polylinestring) != str:
+        return None
+    polyline = polylinestring.split(' ')
+    geometry=[]
+    for i in polyline:
+        coordinates = i.split(',')
+        coords=[]
+        for j in coordinates:
+            coords.append(float(j))
+        geometry.append(coords)
+    return geometry
+
+    
+
+
+
+
+
+# function to get event status
+def get_event_status(start_time_string, end_time_string):
+    start_time = dateutil.parser.parse(start_time_string)
+
+    event_status = "active"
+    if datetime.now(timezone.utc) < start_time:
+        event_status = "planned"  # if < 2 to 3 weeks make it pending instead of planned
+    elif end_time_string:
+        end_time = dateutil.parser.parse(end_time_string)
+
+        if end_time < datetime.now(timezone.utc):
+            event_status = "completed"
+    return event_status
+
+
+
+
+
+
+
+# Parse COtrip alert to WZDx
+def parse_alert(alert, callback_function=None):
+    if not validate_alert(alert):
+        if callback_function:    #Note :a call back fucnction , which will trigger every time the invalid data is given
+            callback_function(alert)
+        return None
+    feature = {}
+    geometry = {}
+    geometry['type'] = "LineString"
+    geometry['coordinates'] = parse_polyline(alert['alert:Polyline'])
+
+    feature['type'] = "Feature"
+    properties = {}
+
+    #### I included a skeleton of the message, fill out all required fields and as many optional fields as you can. Below is a link to the spec page for a road event
+    #### https://github.com/usdot-jpo-ode/jpo-wzdx/blob/master/spec-content/objects/RoadEvent.md
+
+    # road_event_id
+    #### Leave this empty, it will be populated by add_ids
+    properties['road_event_id'] = ''
+
+    # Event Type ['work-zone', 'detour']
+    properties['event_type'] = 'work-zone'
+
+    # data_source_id
+    #### Leave this empty, it will be populated by add_ids
+    properties['data_source_id'] = ''
+
+    # start_date
+    properties['start_date'] = reformat_datetime(alert['alert:ReportedTime'])
+
+    # end_date
```

Comment:
We are using two different access methods on the alert object here, can we stick to one for consistency?

Reply:
This seems to be the case throughout the file, let's try to standardize on an access method

Reply:
Is this referring to alert['event'] vs alert.get('event')?

Reply:
Yes

Reply:
@abinashkonersman any update on this one?

Reply:
updated all to .get()